MonoZelda/Controllers/KeyboardController.cs

```diff
@@ -41,38 +51,66 @@ public KeyboardController(CommandManager commandManager)
         };
     }
 
-    // Properties
     public KeyboardState CurrentKeyboardState { get; private set; }
     public KeyboardState PreviousKeyboardState { get; private set; }
 
     public void Update(GameTime gameTime)
     {
         CurrentKeyboardState = Keyboard.GetState();
+        UpdateDirectionalKeysList();
+
+        if (_pressedDirectionalKeys.Any())
+        {
+            Keys mostRecentKey = _pressedDirectionalKeys.Last();
+            _commandManager.Execute(CommandType.PlayerMoveCommand, mostRecentKey);
+        }
+        else
+        {
+            _commandManager.Execute(CommandType.PlayerStandingCommand, Keys.None);
+        }
 
-        // Iterate keyCommandDictionary to check input
         foreach (var keyCommandPair in _keyCommandDictionary)
         {
             (Keys key, bool oneShot) = keyCommandPair.Key;
-            if (!oneShot && (CurrentKeyboardState.IsKeyDown(key) || key == Keys.None))
+
+            if (_directionalKeys.Contains(key))
+                continue;
+
+            if (!oneShot && CurrentKeyboardState.IsKeyDown(key))
             {
                 _commandManager.Execute(keyCommandPair.Value, key);
                 break;
             }
-            else if(oneShot && OneShotPressed(key))
+            else if (oneShot && OneShotPressed(key))
             {
                 _commandManager.Execute(keyCommandPair.Value, key);
                 break;
             }
         }
 
-        // Update previous keyboard state (Do after all keyboard checks)
         PreviousKeyboardState = CurrentKeyboardState;
     }
 
+    private void UpdateDirectionalKeysList()
+    {
+        foreach (Keys key in _directionalKeys)
+        {
+            bool isCurrentlyPressed = CurrentKeyboardState.IsKeyDown(key);
+            bool wasPressed = PreviousKeyboardState.IsKeyDown(key);
+
+            if (isCurrentlyPressed && !wasPressed)
+            {
+                _pressedDirectionalKeys.Remove(key);
+                _pressedDirectionalKeys.Add(key);
+            }
```

Comment:
This is a pretty clever and simple way to prioritize recently pressed keys. Worth noting that List.Remove(item) does not depend on item being in the list, which is why this works.